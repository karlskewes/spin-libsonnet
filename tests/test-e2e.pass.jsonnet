local spin = import '../spin.libsonnet';

// Simple stack to exercise spin.libsonnet.

local appName = 'e2e';
local email = 'me@example.com';

local app = spin.application(appName, email);

local pipeline = spin.pipeline('deploy', appName);

local project = spin.project('end-to-end', email) + {
  config+: {
    applications+: app.name,
    pipelineConfigs+: [
      {
        application: app.name,
        pipelineConfigId: pipeline.name,
      },

    ],
  },
};

{
  ['application-' + appName]: app,
  ['pipeline-' + appName + '-' + pipeline.name]: pipeline,
  ['project-' + project.name]: project,
}
