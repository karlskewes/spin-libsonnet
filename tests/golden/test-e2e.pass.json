{
   "application-e2e": {
      "cloudProviders": "kubernetes",
      "email": "me@example.com",
      "name": "e2e",
      "user": "me@example.com"
   },
   "pipeline-e2e-deploy": {
      "application": "e2e",
      "keepWaitingPipelines": false,
      "limitConcurrent": true,
      "name": "deploy"
   },
   "project-end-to-end": {
      "config": {
         "applications": "[ ]e2e",
         "clusters": [ ],
         "pipelineConfigs": [
            {
               "application": "e2e",
               "pipelineConfigId": "deploy"
            }
         ]
      },
      "email": "me@example.com",
      "id": "end-to-end",
      "name": "end-to-end"
   }
}
